### ISSUE TITLE
Automatic migrations for signal inputs, outputs and view queries

### ISSUE THREAD

**Issue by eneajaho** [angular/angular#54808]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

Angular is moving towards signal inputs and function-based outputs. Should Angular provide official migrations for these types of migrations? 

In ngxtension library we already have added support for migrating signal inputs and there's a [PR](https://github.com/nartc/ngxtension-platform/pull/301) in the works for the outputs (https://github.com/nartc/ngxtension-platform/pull/301) 

https://ngxtension.netlify.app/utilities/migrations/signal-inputs-migration/

Should we add those migrations to Angular itself, or it's okay if we keep them in a community library. 

### Proposed solution

Migrate the migrations from the community library to Angular itself 

### Alternatives considered

Keep them in community library only.

**Comment by pkozlowski-opensource** [comment]
tl;dr; is that we would _love_ to provide an automatic migration but still debating scope / what can be automatically covered. 

We are actively discussing the topic of migrations for singal-based inputs, outputs and queries. Outputs are fairly straightforward but inputs and queries might be more involved depending on the usage patterns. Some examples:
* with inputs we don't need `ngOnChanges` as a `computed` is a better replacement - should it be migrated automatically?
* queries no longer expose `QueryList` so the usage patterns are potentially different if people subscribe to `changes`

Also, we would love to collaborate with the community on the migration so if anyone is interested in helping out - please let us know!

**Comment by rondo117** [comment]
When do we expect to have the automatic migration?

**Comment by condorthe2nd** [comment]
Is this going to happen?

**Comment by JoostK** [comment]
Yes, in fact it is in full swing: [is:pr label:"area: migrations" signal ](https://github.com/angular/angular/pulls?q=is%3Apr+label%3A%22area%3A+migrations%22+signal)

**Comment by pkozlowski-opensource** [comment]
The good news here that all the mentioned migrations landed as part of v19!

**Comment by pkozlowski-opensource** [comment]
Docs:
* https://angular.dev/reference/migrations/signal-inputs
* https://angular.dev/reference/migrations/signal-queries
